user_interface.cpp

```diff
@@ -0,0 +1,112 @@
+#include <iostream>
+#include <string>
+#include "user_interface.h"
+#include <cmath>
+#include <iomanip> //std::setprecision
+
+using namespace std;
+//const ostream &standard = std::cout;
+const string white_line = "\011\012\013\014\015\040";
+
+void print_greeting(ostream &out)
+{
+    out << "Welcome to the random number generator!\n"
+        << "It was done as a laboratory programming job.\n";
+}
+
+string clear_whites(const string &s){
+    size_t first = s.find_first_not_of(white_line);
+
+    if(first == string::npos) return "";
+    size_t last = s.find_last_not_of(white_line, s.size() );
+    return s.substr(first, last - first + 1);
+}
+
+int choose_gen(ostream &out)
+{
+    bool correct_ans = false;
+    string ans = "";
+    int res = 11;
+    size_t ans_len = 0;
+    do{
+        print_help();
+        out << "Select one of the generators by entering its serial number.\n"
+            << "Enter H to see help again.\n"
+            << "Enter Q to exit the program.\n"
+            << "Hint: All other symbols will call help info and ask you to enter the data again.\n" ;
+        cin >> ans;
+        if(cin.fail()){
+            cin.clear();
+            cin.ignore();
+            throw out_of_range("***** FAILED TO GET STRING FROM STD::CIN *****\nWatch size_t choose_gen(ostream &out) to solve the problem\n");
+        }
+        ans = clear_whites(ans);
+        ans_len = ans.length();
+        if(ans_len > 0 && ans_len < 3){
+            if(ans_len == 2 && ans == "10"){
+                res = 10;
+                correct_ans = true;
+            }
+            else if(ans_len == 1){
+                    if(ans[0] == 'q' || ans[0] == 'Q' || ans[0] == '0'){
+                        res = 0;
+                        correct_ans = true;
+                    }
+                    if(ans[0] >= '0' && ans[0] <= '9'){
+                        res = int(ans[0]) - int('0');
+                        correct_ans = true;
```

Comment:
Just `return ans[0] - '0';`
I feel you don't need to type cast and subtracting will just work. Please, check that. Otherwise keep it as `return int(ans[0]) - int('0');`